src/home_audit.ts

```diff
@@ -0,0 +1,807 @@
+    roofs.forEach(roof => {
+        if (roof.knee_wall) {
+            knee_walls.push(roof.knee_wall)
+        }
+    })
+    const combined_knee_wall_area = getSumOfObjectPropertiesByFieldName(knee_walls, 'area')
+    if(combined_knee_wall_area > max_knee_wall_area) {
+        roofs.forEach((roof, index) => {
+            if(roof.knee_wall && roof.knee_wall.area) {
+                addErrorMessage(`zone/zone_roof/${index}/knee_wall/area`, `Total knee wall area exceeds the maximum allowed ${Math.ceil(max_knee_wall_area)} sqft (2/3 the footprint area).`)
+            }
+        })
+    }
+}
+
+/**
+ * Check that the roof area isn't too big for the roof type
+ */
+function checkRoofArea(
+    floors:Floor[],
+    roofs: Roof[],
+    conditioned_footprint:number,
+    type:'roof_area'|'ceiling_area'
+): void
+{
+    const roof_type = type === 'roof_area' ? 'cath_ceiling' : 'vented_attic'
+    const combined_type = type === 'roof_area' ? 'roof' : 'ceiling'
+    const combined_area_invalid = getRoofCoversFloorErrorMessage(floors, roofs)
+    if(!combined_area_invalid) {
+        const combined_roof_ceil_area = getCombinedArea_roof_and_ceiling(roofs)
+        const conditioned_area_invalid = getConditionedAreaErrorMessage(combined_roof_ceil_area, conditioned_footprint, combined_type)
+        if(conditioned_area_invalid) {
+            roofs.forEach((roof, index) => {
+                if(roof.roof_type === roof_type) {
+                    addErrorMessage(`/zone/zone_roof/${index}/${type}`, conditioned_area_invalid)
+                }
+            })
+        }
+    }
+    else {
+        roofs.forEach((roof, index) => {
+            if(roof.roof_type === roof_type) {
+                addErrorMessage(`/zone/zone_roof/${index}/${type}`, combined_area_invalid)
+            }
+        })
+    }
+}
+
+/**
+ * Check that the floor isn't too small for the combined area
+ */
+function checkFloorArea(floors: Floor[], roofs: Roof[], conditioned_footprint:number): void
+{
+    const combined_area_invalid = getRoofCoversFloorErrorMessage(floors, roofs)
+    if(!combined_area_invalid) {
+        const combined_floor_area = getCombinedArea_floor(floors)
+        const conditioned_area_invalid = getConditionedAreaErrorMessage(combined_floor_area, conditioned_footprint, 'floor')
+        if(conditioned_area_invalid) {
+            floors.forEach((floor, index) => {
+                addErrorMessage(`/zone/zone_floor/${index}/floor_area`, conditioned_area_invalid)
+            })
+        }
+    }
+    else {
+        floors.forEach((roof, index) => {
+            addErrorMessage(`/zone/zone_floor/${index}/floor_area`, combined_area_invalid)
+        })
+    }
+}
+
+/**
+ * Check that the insulation level is appropriate for the foundation type
+ */
+function checkFoundationLevel(floors:Floor[]): void
+{
+    floors.forEach(({foundation_type, foundation_insulation_level}, index) => {
+        if(foundation_type && isNullOrUndefined(foundation_insulation_level)) {
```

Comment:
`foundation_insulation_level` is not required for all home anymore.  See line 880 of _base.schema.json
Test with regression_files/Foundation_type_above_other_unit.json